balancer/weightedtarget/weightedaggregator/aggregator.go

```diff
@@ -192,13 +192,12 @@ func (wbsa *Aggregator) UpdateState(id string, newState balancer.State) {
 		return
 	}
 
-	wbsa.csEvltr.RecordTransition(oldState.stateToAggregate, newState.ConnectivityState)
-
 	if !(oldState.state.ConnectivityState == connectivity.TransientFailure && newState.ConnectivityState == connectivity.Connecting) {
 		// If old state is TransientFailure, and new state is Connecting, don't
 		// update the state, to prevent the aggregated state from being always
 		// CONNECTING. Otherwise, stateToAggregate is the same as
 		// state.ConnectivityState.
+		wbsa.csEvltr.RecordTransition(oldState.stateToAggregate, newState.ConnectivityState)
 		oldState.stateToAggregate = newState.ConnectivityState
```

Comment:
Optional: Naming the variable that stores the current state as `oldState` makes things a little confusing here since we are updating the current state based on the new state. So, things like `oldState.stateToAggregate = newState.ConnectivityState` and `oldState.state = newState` would read much better if as `state.stateToAggregate = newState.ConnectivityState` and `state.state = newState`. What do you think?

Reply:
makes sense to me. The `oldState` was misleading when i read it initially. Il make the update